nextflow_schema.json

```
@@ -2,109 +2,130 @@
+                    "type": "number",
+                    "default": 0.05,
+                    "description": "Confidence in the classification of a read as a certain taxon. Refere to https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown#confidence-scoring for details."
                 }
             }
         },
         "kraken2taxa": {
             "title": "kraken2taxa",
             "type": "object",
-            "description": "",
+            "description": "Used for fine-tuning the isolation step of kraken2 output. Also here the taxon/taxa to be assessed/to be filtered can be set.",
             "default": "",
             "properties": {
+                "cutoff_tax2filter": {
+                    "type": "integer",
+                    "default": 2,
+                    "description": "If a read has less k-mers assigned to the taxon/taxa to be assessed/to be filtered the read is ignored by the pipeline."
+                },
+                "cutoff_unclassified": {
+                    "type": "number",
+                    "default": 0.0,
+                    "description": "If the ratio of the sum of k-mers mapped to the taxon/taxa to be assessed/to be filtered to unclassified is below this threshold the read is ignored by the pipeline. Can take values between 0.0 and 1.0.",
+                    "minimum": 0,
+                    "maximum": 1
+                },
                 "tax2filter": {
                     "type": "string",
-                    "default": "Mammalia"
+                    "default": "Homo",
+                    "description": "The taxon or taxonomic group to be assessed or filtered by the pipeline."
                 }
             }
         },
-        "kraken2": {
-            "title": "kraken2",
+        "blastn": {
+            "title": "blastn",
             "type": "object",
-            "description": "",
+            "description": "Parameters to fine-tune the output of blastn.",
             "default": "",
             "properties": {
-                "kraken2db": {
-                    "type": "string",
-                    "default": "https://genome-idx.s3.amazonaws.com/kraken/k2_standard_08gb_20230605.tar.gz"
+                "blast_coverage": {
+                    "type": "number",
+                    "default": 40.0,
+                    "description": "Coverage is the percentage of the query sequence which can be found in the alignments of the sequence match. It can be used to fine-tune the validation step."
                 },
-                "save_output_fastqs": {
-                    "type": "boolean"
+                "blast_outformat": {
+                    "type": "string",
+                    "default": "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qcovs qcovhsp",
+                    "hidden": true,
+                    "description": "Output format of blastn. Should not be changed!"
```

```suggestion
                    "description": "Output format of blastn. Non default output formats will not be compatible with downstream processes and will not generate a summary!"
```

Did you add safety measures to not crash the pipeline and throw an error instead for the user, in case someone tries to change the blastn output?

---

Otherwise just remove this parameter for now, and patch the blastn module/set the default in the pipeline.nf